module/VuFind/src/VuFind/Db/Table/ResourceTags.php

```diff
@@ -237,10 +237,18 @@ public function destroyLinks($resource, $user, $list = null, $tag = null)
         $callback = function ($select) use ($resource, $user, $list, $tag) {
             $select->where->equalTo('user_id', $user);
             if (null !== $resource) {
-                if (!is_array($resource)) {
-                    $resource = [$resource];
+                if ($resource === 'listTag') {
+                    // special case -- if $resource is set to the string "listTag",
+                    // we want to retrieve tags assigned to a user list
+                    // and filter out user resource tags
+                    // (resource_id is NULL for list tags).
+                    $select->where->isNull('resource_id');
+                } else {
+                    if (!is_array($resource)) {
```

Comment:
While you're changing this code anyway, you could optimize it to:

`$select->where->in('resource_id', (array)$resource);`

and get rid of the preceding `if` statement.